cachito/workers/pkg_managers/pip.py

```diff
@@ -17,6 +17,53 @@
 NOTHING = object()  # A None replacement for cases where the distinction is needed
 
 
+def get_pip_metadata(package_dir):
+    """
+    Attempt to get the name and and version of a Pip package.
+
+    First, try to parse the setup.py script (if present) and extract name and version
+    from keyword arguments to the setuptools.setup() call. If either name or version
+    could not be resolved and there is a setup.cfg file, try to fill in the missing
+    values from metadata.name and metadata.version in the .cfg file.
+
+    If neither setup.py nor setup.cfg provide the metadata we are looking for, fall
+    back to using the basename of the package directory as package name and leave
+    package version unspecified.
+
+    :param str package_dir: Path to the root directory of a Pip package
+    :return: Tuple of strings (name, version), version may be None
+    """
+    name = None
+    version = None
+
+    setup_py = SetupPY(package_dir)
+    setup_cfg = SetupCFG(package_dir)
+
+    if setup_py.exists():
+        log.info("Extracting metadata from setup.py")
+        name = setup_py.get_name()
+        version = setup_py.get_version()
+    else:
+        log.warning("No setup.py in directory, package is likely not Pip compatible")
+
+    if not (name and version) and setup_cfg.exists():
+        log.info("Filling in missing metadata from setup.cfg")
+        name = name or setup_cfg.get_name()
+        version = version or setup_cfg.get_version()
+
+    if not name:
+        log.warning("Falling back to using directory name as package name")
+        name = Path(package_dir).name
+
+    log.info("Resolved package name: %r", name)
+    if version:
+        log.info("Resolved package version: %r", version)
+    else:
+        log.warning("Could not resolve package version")
```

Comment:
Should this function deal with git data? From what I understand, the git ref (which would be the fallback version) will already be present in the request, so this function does not seem like the right place to deal with it.

Reply:
+1; This should be resolved later, when the function is called.